examples/js/FlatShadedGeometry.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @author astrak / astrak.github.io
+ */
+ /*
+ This is the very basics to allow a flat shaded look on THREE.MeshLambertMaterial. The function loops through the faces of an instance of THREE.Geometry and duplicates every vertices, including identical vertices on different faces.
+ */
+
+THREE.FlatShadedGeometry = function ( geometry ) {
+
+	var flatShadedGeometry = geometry.clone();
+
+	var vertices = [], 
+		faces = [],
+		morphTargets = [];
+
+	if ( geometry.morphTargets ) {
+
+		for ( var i = 0 ; i < geometry.morphTargets.length ; i++ ) {
+
+			morphTargets[ i ] = { name : geometry.morphTargets.name, vertices : [] };
+
+		}
+
+	}
+
+	for ( var i = 0 ; i < geometry.faces.length ; i++ ) {
+
+		var f = geometry.faces[ i ];
+
+		vertices[ i * 3 ] = geometry.vertices[ f.a ].clone();
+		vertices[ i * 3 + 1 ] = geometry.vertices[ f.b ].clone();
+		vertices[ i * 3 + 2 ] = geometry.vertices[ f.c ].clone();

```

Comment:
Yes I think I can : as far as i know the problem to solve is that vertices shared between faces that do not have the same normal. Those lines duplicate the vertices of each face so that they surely get this same normal attribute value. And since the face is composed of new vertices with other indices, I clone the face too.

I am just sure that this makes a flat shading, as visible in the linked demo, and that I do not get flat shading without this.

Feel free to tell something wrong or useless or dirty, I just want to share a solution.